### PR TITLE
Move remaining settings in NettyHttpServerTransport to the new infra

### PR DESCRIPTION
**Pull request by bleskes**
Some bw incompatible setting changes (to simplify and conform to other settings):
- http.netty.http.blocking_server -> http.tcp.blocking_server
- http.netty.host (removed, we just have http.host)
- http.netty.bind_host (removed, we just have http.bind_host)
- http.netty.publish_host (removed, we just have http.publish_host)
- http.netty.tcp_no_delay -> http.tcp.no_delay
- http.netty.tcp_keep_alive -> http.tcp.keep_alive
- http.netty.reuse_address -> http.txp.reuse_address
- http.netty.tcp_send_buffer_size -> http.tcp.send_buffer_size
- http.netty.tcp_receive_buffer_size -> http.tcp.receive_buffer_size

### CODE REVIEW

**Comment by bleskes** [comment]
@colings86 care to take a look?

File: core/src/main/java/org/elasticsearch/http/netty/NettyHttpServerTransport.java [hunk 1]
```diff
@@ -117,6 +115,52 @@
...
+                    defaultReceiverPredictor = Math.min(defaultReceiverPredictor, Math.max(l, 64 * 1024));
+                }
+                return new ByteSizeValue(defaultReceiverPredictor).toString();
+            }, false, Setting.Scope.CLUSTER);
+    public static final Setting<ByteSizeValue> SETTING_HTTP_NETTY_RECEIVE_PREDICTOR_MIN = byteSizeSetting("http.netty" +
+            ".receive_predictor_min",
+            SETTING_HTTP_NETTY_RECEIVE_PREDICTOR_SIZE, false, Setting.Scope.CLUSTER);
+    public static final Setting<ByteSizeValue> SETTING_HTTP_NETTY_RECEIVE_PREDICTOR_MAX = byteSizeSetting("http.netty" +
+            ".receive_predictor_max",
+            SETTING_HTTP_NETTY_RECEIVE_PREDICTOR_SIZE, false, Setting.Scope.CLUSTER);
```
Comment (colings86) [inline core/src/main/java/org/elasticsearch/http/netty/NettyHttpServerTransport.java]: Should all these constants be in HttpTransportSettings with the other settings?

**Comment by colings86** [comment]
@bleskes I left a comment about where the constants should live but otherwise LGTM

File: core/src/main/java/org/elasticsearch/common/settings/ClusterSettings.java [hunk 2]
```diff
@@ -150,240 +152,253 @@ public void apply(Settings value, Settings current, Settings previous) {
...
-            PageCacheRecycler.TYPE_SETTING,
-            PluginsService.MANDATORY_SETTING,
-            BootstrapSettings.SECURITY_MANAGER_ENABLED_SETTING,
-            BootstrapSettings.SECURITY_FILTER_BAD_DEFAULTS_SETTING,
-            BootstrapSettings.MLOCKALL_SETTING,
-            BootstrapSettings.SECCOMP_SETTING,
-            BootstrapSettings.CTRLHANDLER_SETTING
-        )));
+            Arrays.asList(AwarenessAllocationDecider.CLUSTER_ROUTING_ALLOCATION_AWARENESS_ATTRIBUTE_SETTING,
+                    TransportClientNodesService.CLIENT_TRANSPORT_NODES_SAMPLER_INTERVAL, // TODO these transport client settings are kind
```
Comment (s1monw) [inline core/src/main/java/org/elasticsearch/common/settings/ClusterSettings.java]: why moving from 4 to 8 spaces?

File: core/src/main/java/org/elasticsearch/http/netty/NettyHttpServerTransport.java (hunk 1 above)
Comment (bleskes) [inline core/src/main/java/org/elasticsearch/http/netty/NettyHttpServerTransport.java]: HttpTransportSettins is used from the security class so I thought we should keep it to the bare logical set that is needed. i put all of these here, but I dound that the host setting , although not needed by security.java, belongs next to port.

File: core/src/main/java/org/elasticsearch/common/settings/ClusterSettings.java (hunk 2 above)
Comment (bleskes) [inline core/src/main/java/org/elasticsearch/common/settings/ClusterSettings.java]: I have no idea - standard intellij indentation for me. Here is what I have - is it different for you?

image

Comment (s1monw) [inline core/src/main/java/org/elasticsearch/common/settings/ClusterSettings.java]: same!

Comment (bleskes) [inline core/src/main/java/org/elasticsearch/common/settings/ClusterSettings.java]: Then I have no idea why my intellij have changed this while others kept it the same... meh :(

**Pull request closed without merging**